src/disco/FixedWidthField/Real/ENDF/src/write.hpp

```diff
@@ -10,22 +10,88 @@ write( Representation real, Iterator& it ){
   } else if ( isInvalid( real ) ){
     writeInvalid( it );
   } else {
-    std::array< char, w + 3 > buffer;
-    auto greedyExponent = requiresTwoDigitExponent( real );
-    const auto length = w + 2;
-    const auto decimalDigits = 6 - greedyExponent;
-    int consumedCharacters = snprintf( &buffer[0], length + 1, "%*.*E",
-                                       length, decimalDigits, real );
-    if ( consumedCharacters < 0 ){
-      throw std::exception();
+
+    // characters excluded for fixed notation precision (initial space, digit,
+    // decimal point)
+    const unsigned int NUMBER_EXCLUDED_CHARS = 3;
+
+    // roundoff limit
+    const double ROUNDOFF_LIMIT = 0.5;
+
+    // Minimal exponent size is 2 (one sign and one digit)
+    constexpr int MIN_EXPWIDTH = 2;
+    // Minimal exponent for which we can use fixed notation
+    constexpr int MIN_EXPONENT = -2;
+
+    // Decompose real
+    double significand{ real };
+    int exponent{ 0 };
+    int expWidth{ MIN_EXPWIDTH };
+    if( real != 0.0 ){
+      // log10(significand 10^exponent) = exponent + log10(significand) and
+      // log10(significand) is within [0,1] so that exponent is given by the
+      // floor value
+      exponent = static_cast< int >( 
+        std::floor( std::log10( std::abs( real ) ) ) );
+
+      if( 0 != exponent ){
+        significand /= std::pow( 10.0, exponent );
+        expWidth += static_cast< int >( 
+            std::floor( std::log10( std::abs( exponent ) ) ) );
+      }
+      // if( 0 < exponent ){
+      //   significand /= std::pow( 10.0, exponent );
+      //   expWidth += static_cast< int >( 
+      //       std::floor( std::log10( std::abs( exponent ) ) ) );
+      // } else if( 0 > exponent ){
+      //   significand *= std::pow( 10.0, -exponent );
+      //   expWidth += static_cast< int >( 
+      //       std::floor( std::log10( std::abs( exponent ) ) ) );
+      // }
+    }
+
+    unsigned int width = w - expWidth;
```

Comment:
`w` is such a great variable name, especially one not locally scoped 🙄

Reply:
Yeah, I didn't pick that name. I tried to change it, but it had a larger effect than what I was wiling to play around with.